src/modules/trade/pure/TradeNumberInput/index.tsx

```diff
@@ -0,0 +1,62 @@
+import { TradeWidgetField, TradeWidgetFieldProps } from 'modules/trade/pure/TradeWidgetField'
+import { useCallback, useEffect, useState } from 'react'
+import { NumericalInput, Suffix } from './styled'
+
+export interface TradeNumberInputProps extends TradeWidgetFieldProps {
+  value: number | null
+  onUserInput(input: number | null): void
+  suffix?: string
+  decimalsPlaces?: number
+  min?: number
+  max?: number
+}
+
+export function TradeNumberInput(props: TradeNumberInputProps) {
+  const { value, suffix, onUserInput, decimalsPlaces = 0, min, max = 0 } = props
+
+  const [displayedValue, setDisplayedValue] = useState(value === null ? '' : value.toString())
+
+  const onChange = useCallback(
+    (newValue: string) => {
+      const hasDot = newValue.includes('.')
+      const [quotient, decimals] = (newValue || '').split('.')
+      const filteredDecimals = decimalsPlaces && hasDot ? `.${decimals.slice(0, decimalsPlaces)}` : ''
+      const adjustedValue = quotient + filteredDecimals
+      const parsedValue = adjustedValue ? parseFloat(adjustedValue) : null
+
+      if (parsedValue && parsedValue > max) {
+        setDisplayedValue(max.toString())
+        onUserInput(max)
+        return
+      }
+
+      if ((parsedValue && min && parsedValue < min) || !parsedValue) {
+        setDisplayedValue(min?.toString() || '')
+        onUserInput(min || null)
+        return
+      }
+
+      setDisplayedValue(adjustedValue)
+
+      if (value !== parsedValue) {
+        onUserInput(parsedValue)
+      }
+    },
+    [onUserInput, value, min, max, decimalsPlaces]
+  )
+
+  useEffect(() => {
+    onChange(value ? value.toString() : '')
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
```

Comment:
In this case, I realized you can just not provide the empty array and eslint won't complain

```suggestion
  })
```

Reply:
This is an initial setup of value, added comment

Reply:
Yes, not a problem.
I'm just saying you don't need the empty param array (`[]`) nor the eslint rule disabling